README.md

```diff
@@ -26,5 +29,87 @@ pyqasm requires Python 3.10 or greater, and can be installed with pip as follows
 pip install pyqasm
 ```
 
->[!WARNING]
-> **This project is "pre-alpha", and is not yet stable or fully realized. Use with caution, as the API and functionality are subject to significant changes.**
+## Install from source 
+
+You can also install from source by cloning this repository and running a pip install command
+in the root directory of the repository:
+
+```shell
+git clone https://github.com/qBraid/pyqasm.git
+cd pyqasm
+pip install .
+```
+
+## Check version
+
+You can view the version of pyqasm you have installed within a Python shell as follows:
+
+```python
+import pyqasm
+print(pyqasm.__version__)
+```
+
+## Usage Examples
+
+### Unrolling OpenQASM 3 program 
+
+```python
+from pyqasm.unroller import unroll
+
+program = """
+OPENQASM 3;
+include "stdgates.inc";
+
+qubit[2] q;
+bit[2] c;
+
+h q[0];
+cx q[0], q[1];
+
+measure q->c;
+"""
+
+unrolled_qasm = unroll(program).unrolled_qasm
+print(unrolled_qasm)
+```
```

Comment:
Let's write import like this, a little cleaner
 
```python
import pyqasm

...

unrolled_qasm = pyqasm.unroll(program)
```

Also, can we make it so that `pyqasm.unroll` returns the unrolled qasm string directly intead of a `Qasm3Module`? Unless there's a good reason to keep it in that object.

Also, correct me if I'm wrong, but I think that

```
measure q -> c;
```

is OpenQASM 2 syntax, and is still just supported by OpenQASM 3 for backwards compatibility. I'm pretty sure the new syntax is something like

```
c[0] = measure q[0]; // single qubit
c = measure q;  // multiple qubits
```

I know that's correct for single, but might wanna double check for multiple

Reply:
Yea so `qbraid-qir` currently expects an `openqasm3.ast.Program` object which is present as one of the attributes of `Qasm3Module`. We can have an arg like - 
```python
def unroll(program, return_type : str = "str"):
   # return_type is one of {"str", "ast"} 
``` 
which just returns the repr. 

The only thing to think about is that qir programs require the total num of [qubits and clbits to be declared at the start](https://github.com/qBraid/qbraid-qir/blob/4b0df921e8831bf31d2d0630da52e686f20adb18/qbraid_qir/qasm3/visitor.py#L105) and if we just return the repr, we lose that info and would likely have to calculate that again.

Reply:
Also, the measure format `c = measure q;` works for multiple too, thanks for the suggestion!

Reply:
Ok gotcha. Two options in my mind:

1. We could declare a function `pyqasm._unroll()` that returns the `Qasm3Module` that we use internally within qBraid-QIR, and then have `pyqasm.unroll()` be the user facing function that returns the string using `pyqasm._unroll()` as a helper.
2. Add a keyword argument `as_module=False` (or something like that) to the function so that by default, `pyqasm.unroll()` returns a string, by within qBraid-QIR, we can specify `pyqasm.unroll(..., as_module=True)` to keep it in the form of a `Qasm3Module`

Either way works, whichever you think is best

Reply:
Second one works, will update the code. Thanks!